### ISSUE TITLE
aws-autoscaling: The specified instance m7i.xlarge is not valid

### ISSUE THREAD

**Issue by peterrsongg** [aws/aws-cdk#28459]
### Describe the bug

When trying to create an autoscaling group with instance type `m7i.xlarge` and machineImage ` ec2.MachineImage.latestAmazonLinux2023()` I receive the following error:
```
 Deployment failed: Error: The stack named Dev-PerfTests failed creation, 
it may need to be manually deleted from the AWS console: ROLLBACK_COMPLETE: 
Resource handler returned message: "The specified instance type m7i.xlarge is not valid
 (Service: AutoScaling, Status Code: 400, Request ID: 0d9c9cd9-d2fe-48c7-98f2-ee04a8f93602)"
 (RequestToken: 70bffecd-981f-7ad2-8ebd-f0d7d8730f02, HandlerErrorCode: InvalidRequest),
 Resource handler returned message: "The specified instance type m7i.large is not valid
 (Service: AutoScaling, Status Code: 400, Request ID: 8404beeb-4b40-4afc-878b-8dfd1d3a787f)"
 (RequestToken: 2c522a9e-440f-965a-aca9-3252ef99729e, HandlerErrorCode: InvalidRequest)
```

### Expected Behavior

The autoscaling group is able to be created with m7i.xlarge

### Current Behavior

The autoscaling group fails creation with the above error. Interestingly, this same configuration will work in aws-cdk-lib/ec2 and m6i will work if specified in the autoscaling group, but m7I does not

### Reproduction Steps

Create an autoscaling group like so:
```
    const vpc = ec2.Vpc.fromLookup(this, 'VPC', { isDefault: true });
    // lets create a security group for our instance
    // A security group acts as a virtual firewall for your instance to control inbound and outbound traffic.
    const securityGroup = new ec2.SecurityGroup(this, 'simple-instance-1-sg', {
      vpc: vpc,
      allowAllOutbound: true, // will let your instance send outboud traffic
      securityGroupName: 'simple-instance-1-sg',
    });

    const role = new Role(this, 'ec2Role', {
      assumedBy: new ServicePrincipal('ec2.amazonaws.com'),
    });
    role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AWSCodeCommitReadOnly'));
    role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('CloudWatchFullAccess'));
    role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore'));

    const autoScalingGroupA = new AutoScalingGroup(this, 'AutoScalingGroupA', {
      vpc: vpc,
      instanceType: ec2.InstanceType.of(ec2.InstanceClass.M6I, ec2.InstanceSize.XLARGE),
      machineImage: ec2.MachineImage.latestAmazonLinux2023(),
      allowAllOutbound: true,
      maxCapacity: 1,
      minCapacity: 0,
      desiredCapacity: 0,
      healthCheck: HealthCheck.ec2(),
      securityGroup: securityGroup,
      role: role,
    });

```

### Additional Information/Context

aws-cdk-lib version: 2.115.0
constructs: 10.3.0

### CDK CLI Version

2.115

### Node.js Version

14.21.3

### OS

Amazon Linux

### Language

TypeScript

### Language Version

Typescript (5.0.4)

**Comment by pahud** [comment]
Are you able to select that instance type from AWS EC2 console?
Which region are you deploying to?
I am afraid the instance type might not be available in your region.

I am able to deploy m7x.large in `us-east-1` with the CDK code below:


```ts

// get autoscaling group
export function getAutoScalingGroup(scope: Construct, instanceType?: ec2.InstanceType, machineImage?: ec2.MachineImage): autoscaling.AutoScalingGroup {
    return new autoscaling.AutoScalingGroup(scope, 'AutoScalingGroup', {
        instanceType: instanceType ?? ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.LARGE),
        machineImage: new ec2.AmazonLinuxImage({ generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2023 }),
        vpc: getDefaultVpc(scope),
    })
}

export class DemoStack extends MyStack {
  constructor(scope: Construct, id: string, props: StackProps) {
    super(scope, id, props);

   getAutoScalingGroup(this, new ec2.InstanceType('m7i.xlarge'))
  }
}
```

And I saw this in my synthesized launch template

```json
  "AutoScalingGroupLaunchTemplateCE2B3AFE": {
   "Type": "AWS::EC2::LaunchTemplate",
   "Properties": {
    "LaunchTemplateData": {
     "IamInstanceProfile": {
      "Arn": {
       "Fn::GetAtt": [
        "AutoScalingGroupInstanceProfile342FAC7C",
        "Arn"
       ]
      }
     },
     "ImageId": {
      "Ref": "SsmParameterValueawsserviceamiamazonlinuxlatestal2023amikernel61x8664C96584B6F00A464EAD1953AFF4B05118Parameter"
     },
     "InstanceType": "m7i.xlarge",
     "Monitoring": {
      "Enabled": false
     },
```

**Comment by MADDY312** [comment]
I am trying to switch from t2.small to c7a.medium/m7a.medium, It throws the same error, but when trying to switch from t2.small to c5a.large, it works fine. And my region is us-east-1. Has anybody found any solution to this?

**Comment by logrus1** [comment]
I ran into this issue today and couldn't find the solution anywhere.  Wish they made the error more clear.  For me it was because I was still using  AutoScaling Launch configurations instead of launch templates.  

"EC2 Auto Scaling no longer supports new EC2 features or new EC2 instance types released after December 31, 2022 through launch configurations. We recommended that you use launch templates to create new Auto Scaling groups and migrate any existing launch configurations to launch templates."

I use cloud formation so I had to replace my AWS::AutoScaling::LaunchConfiguration resource with an AWS::EC2::LaunchTemplate resource and update my autoscaling group.  So not an issue with the cdk,  just an issue of trying to create/update a deprecated resource usint an instance type newer than 2022.
